packages/node-core/src/indexer/dictionary.service.ts

```diff
@@ -120,6 +123,11 @@ function buildDictQueryFragment(
       first: batchSize.toString(),
     },
   };
+
+  if (useDistinct) {
+    node.args.distinct = ['BLOCK_HEIGHT'];
```

Comment:
I think if we useDistinct, we no longer need to have 
```
lessThan: `"${queryEndBlock}"`,
```

if the distinct blocks in between difference is large than queryEndBlock threshold 10000, this will become inefficient. 

For example expected distinct return 1999, 8999, 28999... (total number of batch size blocks)
but due to query limit 

        blockHeight: {
          greaterThanOrEqualTo: `1999`,
          lessThan: `11999`,
        },

we will only have 2 blocks in return

Reply:
I think we need to retain `lessThan` for the work that @bz888 has done. But thinking about that, the `queryEndBlock` could be set to the height when a next dataSource is introduced. 

Maybe we should save this for another task?

Reply:
Ok, let's do it in another task

Reply:
https://github.com/subquery/subql/issues/1381